src/algorithms/KirlikSayin.jl

```diff
@@ -99,68 +100,62 @@ function optimize_multiobjective!(algorithm::KirlikSayin, model::Optimizer)
     variables = MOI.get(model.inner, MOI.ListOfVariableIndices())
     n = MOI.output_dimension(model.f)
     yI, yN = zeros(n), zeros(n)
-    δ = sense == MOI.MIN_SENSE ? -1 : 1
+    # This tolerance is really important!
+    δ = 1.0
     scalars = MOI.Utilities.scalarize(model.f)
     # Ideal and Nadir point estimation
     for (i, f_i) in enumerate(scalars)
+        # Ideal point
         MOI.set(model.inner, MOI.ObjectiveFunction{typeof(f_i)}(), f_i)
-        MOI.set(model.inner, MOI.ObjectiveSense(), sense)
         MOI.optimize!(model.inner)
         status = MOI.get(model.inner, MOI.TerminationStatus())
         if !_is_scalar_status_optimal(status)
             return status, nothing
         end
         _, Y = _compute_point(model, variables, f_i)
-        yI[i] = Y + 1
-        model.ideal_point[i] = Y
-        MOI.set(
-            model.inner,
-            MOI.ObjectiveSense(),
-            sense == MOI.MIN_SENSE ? MOI.MAX_SENSE : MOI.MIN_SENSE,
-        )
+        model.ideal_point[i] = yI[i] = Y
+        # Nadir point
+        MOI.set(model.inner, MOI.ObjectiveSense(), MOI.MAX_SENSE)
         MOI.optimize!(model.inner)
         status = MOI.get(model.inner, MOI.TerminationStatus())
         if !_is_scalar_status_optimal(status)
-            _warn_on_nonfinite_anti_ideal(algorithm, sense, i)
+            # Repair ObjectiveSense before exiting
+            MOI.set(model.inner, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+            _warn_on_nonfinite_anti_ideal(algorithm, MOI.MIN_SENSE, i)
             return status, nothing
         end
         _, Y = _compute_point(model, variables, f_i)
-        yN[i] = Y
+        yN[i] = Y + δ
```

Comment:
So the `+1` (now `+  δ`) was needed because we have a separate constraint that `f_i <= u - δ`. So the box bounds need to account for that?